chat.js

```diff
@@ -456,6 +456,27 @@ class CommandContext {
 		return true;
 	}
 
+	checkFlooding(room, user) {
+		if (!room || !room.filterFlooding || !room.chatLog || user.can('bypassall')) return true;
+		const latestLogs = 4;
+
+		// get alts and previous names
+		const knownNames = [
+			...user.getAltUsers(true),
+			...Object.keys(user.prevNames),
+			user.userid,
+		].map(toId);
+
+		let count = 0;
+		for (let log of room.chatLog.slice(-latestLogs)) {
+			const logUserId = toId(log.split('|')[3]);
+			if (knownNames.includes(logUserId)) count++;
+		}
+
+		if (count >= latestLogs) return this.errorReply(`This room does not allow flooding - you cannot send more than ${latestLogs} message${Chat.plural(latestLogs)} in a row to this room.`);
```

Comment:
this implementation kinda means it's impossible to send 4 lines in a row even in a room that's like 4 hours dead.

Reply:
although, this does make it more difficult for jenny to hit help room, so that's good too...

Reply:
That's the intention

Reply:
>this implementation kinda means it's impossible to send 4 lines in a row even in a room that's like 4 hours dead.

This is a big issue.

Reply:
a way of meeting at the middle for this could be: 
- logging when the last chat message was sent to that specific room.
- if it's more than a minute - the user can post again, otherwise they're capped at 4 messages for that IP.

Instead of logging the message down, maybe put the counter in chat.js whenever a message is accepted and sent through? (Would searching through alts every time you post a message be kinda overkill for CPU usage, specially if users try to have many alts with prevnames, specially when there are easier ways to do that) There, you can put the IP of the connection that sent it in the room's "last 6 talking IPs". Then, if 3 or 4 or more (idk what the best benchmark is) belongs to the user trying to talk in that room, and the room hasn't been dead for more than a minute, reject the message.

I'm sure this probably won't be the best way, but this would cover both the jenny concern, while not killing a room or preventing a roomvoice from speaking for too long.